Compile error in 5.5.0-bbfae0c and above

##### Minetest version
```
5.5.0-c82ec8b (affects 5.5.0-bbfae0c and above)
```

##### OS / Hardware
Operating system: Ubuntu 16.04
CPU: i7-3740QM
GCC: Ubuntu 5.4.0-6ubuntu1~16.04.12

##### Summary

Compiling Minetest crashes with the following log details:

cmake output:
`cmake . -DRUN_IN_PLACE=FALSE -DBUILD_SERVER=TRUE`
```
-- *** Will build version 5.5.0-dev ***
-- Using user-provided IrrlichtMt at subdirectory 'lib/irrlichtmt'
-- *** Building IrrlichtMt 1.9.0 ***
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so  
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.2.54") 
-- Looking for _IRR_COMPILE_WITH_OGLES1_
-- Looking for _IRR_COMPILE_WITH_OGLES1_ - not found
-- Looking for _IRR_COMPILE_WITH_OGLES2_
-- Looking for _IRR_COMPILE_WITH_OGLES2_ - not found
-- Looking for _IRR_COMPILE_WITH_OPENGL_
-- Looking for _IRR_COMPILE_WITH_OPENGL_ - found
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so  
-- Looking for _IRR_LINUX_X11_XINPUT2_
-- Looking for _IRR_LINUX_X11_XINPUT2_ - not found
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/x86_64-linux-gnu/libX11.so
You have called ADD_LIBRARY for library IrrlichtMt without any source files. This typically indicates a problem with your CMakeLists.txt file
-- Using GMP provided by system.
-- Found GMP: /usr/lib/x86_64-linux-gnu/libgmp.so  
-- Using JsonCpp provided by system.
-- Found Json: /usr/lib/x86_64-linux-gnu/libjsoncpp.so  
-- Found LuaJIT: /usr/lib/x86_64-linux-gnu/libluajit-5.1.so  
-- Using LuaJIT provided by system.
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so  
-- cURL support enabled.
-- Found GettextLib: /usr/include  
-- GetText enabled; locales found: ms;ro;th;zh_TW;jbo;el;be;nn;sv;eu;ar;es;pt;zh_CN;fr;kn;pt_BR;cs;ru;mr;it;he;tt;sr_Latn;kk;lv;tr;gl;gd;sl;fi;ko;ky;id;ca;et;ms_Arab;uk;hu;sw;vi;hi;sk;sr_Cyrl;da;ja;de;dv;fil;eo;nb;pl;lt;bg;nl
-- Found OpenAL: /usr/lib/x86_64-linux-gnu/libopenal.so  
-- Found Vorbis: /usr/include  
-- Sound enabled.
-- Found Freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so  
-- Freetype enabled.
-- Looking for cbreak in /usr/lib/x86_64-linux-gnu/libncursesw.so
-- Looking for cbreak in /usr/lib/x86_64-linux-gnu/libncursesw.so - found
-- Found Ncursesw: /usr/lib/x86_64-linux-gnu/libncursesw.so  
-- ncurses console enabled.
-- PostgreSQL not found!
-- LevelDB backend enabled.
-- Redis not found!
-- Found SQLite3: /usr/lib/x86_64-linux-gnu/libsqlite3.so  
-- Prometheus client disabled.
-- SpatialIndex not found!
-- Looking for ZSTD_initCStream
-- Looking for ZSTD_initCStream - found
-- Found Zstd: /usr/lib/x86_64-linux-gnu/libzstd.so  
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for include file endian.h
-- Looking for include file endian.h - found
-- Locale blacklist applied; Locales used: ms;ro;zh_TW;jbo;el;be;nn;sv;eu;es;pt;zh_CN;fr;pt_BR;cs;ru;mr;it;tt;sr_Latn;kk;lv;tr;gl;gd;sl;fi;ko;ky;id;ca;et;uk;hu;sw;vi;sk;sr_Cyrl;da;ja;de;fil;eo;nb;pl;lt;bg;nl
-- Performing Test IS_AARCH64
-- Performing Test IS_AARCH64 - Failed
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") 
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/minetest
```

make output:
```
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/server.cpp.o
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >; _Args = {unsigned int&, SingleMediaDownloader*&}; _Tp = std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >]':
/usr/include/c++/5/bits/alloc_traits.h:530:4:   required from 'static void std::allocator_traits<std::allocator<_Tp1> >::construct(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*, _Args&& ...) [with _Up = std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >; _Args = {unsigned int&, SingleMediaDownloader*&}; _Tp = std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> > >]'
/usr/include/c++/5/bits/vector.tcc:96:30:   required from 'void std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {unsigned int&, SingleMediaDownloader*&}; _Tp = std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >; _Alloc = std::allocator<std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> > >]'
/tmp/minetest/src/network/clientpackethandler.cpp:1563:58:   required from here
/usr/include/c++/5/ext/new_allocator.h:120:4: error: no matching function for call to 'std::pair<unsigned int, std::unique_ptr<SingleMediaDownloader> >::pair(unsigned int&, SingleMediaDownloader*&)'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:206:9: note: candidate: template<class ... _Args1, long unsigned int ..._Indexes1, class ... _Args2, long unsigned int ..._Indexes2> std::pair<_T1, _T2>::pair(std::tuple<_Args1 ...>&, std::tuple<_Args2 ...>&, std::_Index_tuple<_Indexes1 ...>, std::_Index_tuple<_Indexes2 ...>)
         pair(tuple<_Args1...>&, tuple<_Args2...>&,
         ^
/usr/include/c++/5/bits/stl_pair.h:206:9: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h:120:4: note:   mismatched types 'std::tuple<_Args1 ...>' and 'unsigned int'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:155:9: note: candidate: template<class ... _Args1, class ... _Args2> std::pair<_T1, _T2>::pair(std::piecewise_construct_t, std::tuple<_Args1 ...>, std::tuple<_Args2 ...>)
         pair(piecewise_construct_t, tuple<_Args1...>, tuple<_Args2...>);
         ^
/usr/include/c++/5/bits/stl_pair.h:155:9: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h:120:4: note:   cannot convert 'std::forward<unsigned int&>((* & __args#0))' (type 'unsigned int') to type 'std::piecewise_construct_t'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:150:12: note: candidate: template<class _U1, class _U2, class> constexpr std::pair<_T1, _T2>::pair(std::pair<_U1, _U2>&&)
  constexpr pair(pair<_U1, _U2>&& __p)
            ^
/usr/include/c++/5/bits/stl_pair.h:150:12: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h:120:4: note:   mismatched types 'std::pair<_T1, _T2>' and 'unsigned int'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:144:12: note: candidate: template<class _U1, class _U2, class> constexpr std::pair<_T1, _T2>::pair(_U1&&, _U2&&)
  constexpr pair(_U1&& __x, _U2&& __y)
            ^
/usr/include/c++/5/bits/stl_pair.h:144:12: note:   template argument deduction/substitution failed:
/usr/include/c++/5/bits/stl_pair.h:141:38: error: no type named 'type' in 'struct std::enable_if<false, void>'
       template<class _U1, class _U2, class = typename
                                      ^
/usr/include/c++/5/bits/stl_pair.h:138:12: note: candidate: template<class _U2, class> constexpr std::pair<_T1, _T2>::pair(const _T1&, _U2&&)
  constexpr pair(const _T1& __x, _U2&& __y)
            ^
/usr/include/c++/5/bits/stl_pair.h:138:12: note:   template argument deduction/substitution failed:
/usr/include/c++/5/bits/stl_pair.h:136:27: error: no type named 'type' in 'struct std::enable_if<false, void>'
       template<class _U2, class = typename
                           ^
/usr/include/c++/5/bits/stl_pair.h:133:12: note: candidate: template<class _U1, class> constexpr std::pair<_T1, _T2>::pair(_U1&&, const _T2&)
  constexpr pair(_U1&& __x, const _T2& __y)
            ^
/usr/include/c++/5/bits/stl_pair.h:133:12: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h:120:4: note:   cannot convert 'std::forward<SingleMediaDownloader*&>((* & __args#1))' (type 'SingleMediaDownloader*') to type 'const std::unique_ptr<SingleMediaDownloader>&'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:128:17: note: candidate: constexpr std::pair<_T1, _T2>::pair(std::pair<_T1, _T2>&&) [with _T1 = unsigned int; _T2 = std::unique_ptr<SingleMediaDownloader>]
       constexpr pair(pair&&) = default;
                 ^
/usr/include/c++/5/bits/stl_pair.h:128:17: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/5/bits/stl_pair.h:124:12: note: candidate: template<class _U1, class _U2, class> constexpr std::pair<_T1, _T2>::pair(const std::pair<_U1, _U2>&)
  constexpr pair(const pair<_U1, _U2>& __p)
            ^
/usr/include/c++/5/bits/stl_pair.h:124:12: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/5/bits/c++allocator.h:33:0,
                 from /usr/include/c++/5/bits/allocator.h:46,
                 from /usr/include/c++/5/list:61,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/ext/new_allocator.h:120:4: note:   mismatched types 'const std::pair<_T1, _T2>' and 'unsigned int'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
In file included from /usr/include/c++/5/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/5/list:60,
                 from /tmp/minetest/src/environment.h:32,
                 from /tmp/minetest/src/client/clientenvironment.h:22,
                 from /tmp/minetest/src/client/client.h:22,
                 from /tmp/minetest/src/network/clientpackethandler.cpp:20:
/usr/include/c++/5/bits/stl_pair.h:112:26: note: candidate: constexpr std::pair<_T1, _T2>::pair(const _T1&, const _T2&) [with _T1 = unsigned int; _T2 = std::unique_ptr<SingleMediaDownloader>]
       _GLIBCXX_CONSTEXPR pair(const _T1& __a, const _T2& __b)
                          ^
/usr/include/c++/5/bits/stl_pair.h:112:26: note:   no known conversion for argument 2 from 'SingleMediaDownloader*' to 'const std::unique_ptr<SingleMediaDownloader>&'
/usr/include/c++/5/bits/stl_pair.h:108:26: note: candidate: constexpr std::pair<_T1, _T2>::pair() [with _T1 = unsigned int; _T2 = std::unique_ptr<SingleMediaDownloader>]
       _GLIBCXX_CONSTEXPR pair()
                          ^
/usr/include/c++/5/bits/stl_pair.h:108:26: note:   candidate expects 0 arguments, 2 provided
[ 73%] Building CXX object src/CMakeFiles/minetest.dir/network/serverpackethandler.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetest.dir/network/socket.cpp.o
src/CMakeFiles/minetest.dir/build.make:2558: recipe for target 'src/CMakeFiles/minetest.dir/network/clientpackethandler.cpp.o' failed
make[2]: *** [src/CMakeFiles/minetest.dir/network/clientpackethandler.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/serverenvironment.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/serverlist.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/settings.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/staticobject.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/terminal_chat_console.cpp.o
[ 73%] Building CXX object src/CMakeFiles/minetestserver.dir/texture_override.cpp.o
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/tileanimation.cpp.o
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/tool.cpp.o
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/translation.cpp.o
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/version.cpp.o
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/voxel.cpp.o
CMakeFiles/Makefile2:472: recipe for target 'src/CMakeFiles/minetest.dir/all' failed
make[1]: *** [src/CMakeFiles/minetest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 74%] Building CXX object src/CMakeFiles/minetestserver.dir/voxelalgorithms.cpp.o
[ 74%] Linking CXX executable ../bin/minetestserver
[ 74%] Built target minetestserver
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

##### Steps to reproduce
1. Compile Minetest on my old system :P

--- a/src/client/client.h
+++ b/src/client/client.h
@@ -548,7 +548,7 @@ private:
 	// Set of media filenames pushed by server at runtime
 	std::unordered_set<std::string> m_media_pushed_files;
 	// Pending downloads of dynamic media (key: token)
-	std::vector<std::pair<u32, std::unique_ptr<SingleMediaDownloader>>> m_pending_media_downloads;
+	std::vector<std::pair<u32, std::shared_ptr<SingleMediaDownloader>>> m_pending_media_downloads;
 
 	// time_of_day speed approximation for old protocol
 	bool m_time_of_day_set = false;

--- a/src/network/clientpackethandler.cpp
+++ b/src/network/clientpackethandler.cpp
@@ -43,6 +43,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "tileanimation.h"
 #include "gettext.h"
 #include "skyparams.h"
+#include <memory>
+#include <utility>
 
 void Client::handleCommand_Deprecated(NetworkPacket* pkt)
 {
@@ -1559,8 +1561,9 @@ void Client::handleCommand_MediaPush(NetworkPacket *pkt)
 	m_media_pushed_files.insert(filename);
 
 	// create a downloader for this file
-	auto downloader = new SingleMediaDownloader(cached);
-	m_pending_media_downloads.emplace_back(token, downloader);
+	std::unique_ptr<SingleMediaDownloader> downloader{
+		new SingleMediaDownloader(cached) };
+	m_pending_media_downloads.emplace_back(token, std::move(downloader));
 	downloader->addFile(filename, raw_hash);
 	for (const auto &baseurl : m_remote_media_servers)
 		downloader->addRemoteServer(baseurl);

--- a/src/network/clientpackethandler.cpp
+++ b/src/network/clientpackethandler.cpp
@@ -44,7 +44,6 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "gettext.h"
 #include "skyparams.h"
 #include <memory>
-#include <utility>
 
 void Client::handleCommand_Deprecated(NetworkPacket* pkt)
 {
@@ -1561,9 +1560,8 @@ void Client::handleCommand_MediaPush(NetworkPacket *pkt)
 	m_media_pushed_files.insert(filename);
 
 	// create a downloader for this file
-	std::unique_ptr<SingleMediaDownloader> downloader{
-		new SingleMediaDownloader(cached) };
-	m_pending_media_downloads.emplace_back(token, std::move(downloader));
+	auto downloader(std::make_shared<SingleMediaDownloader>(cached));
+	m_pending_media_downloads.emplace_back(token, downloader);
 	downloader->addFile(filename, raw_hash);
 	for (const auto &baseurl : m_remote_media_servers)
 		downloader->addRemoteServer(baseurl);

--- a/src/util/png.cpp
+++ b/src/util/png.cpp
@@ -37,11 +37,11 @@ static void writeChunk(std::ostringstream &target, const std::string &chunk_str)
 
 std::string encodePNG(const u8 *data, u32 width, u32 height, s32 compression)
 {
-	std::ostringstream file{ std::ios::binary };
+	std::ostringstream file(std::ios::binary);
 	file << "\x89PNG\r\n\x1a\n";
 
 	{
-		std::ostringstream IHDR{ std::ios::binary };
+		std::ostringstream IHDR(std::ios::binary);
 		IHDR << "IHDR";
 		writeU32(IHDR, width);
 		writeU32(IHDR, height);
@@ -51,9 +51,9 @@ std::string encodePNG(const u8 *data, u32 width, u32 height, s32 compression)
 	}
 
 	{
-		std::ostringstream IDAT{ std::ios::binary };
+		std::ostringstream IDAT(std::ios::binary);
 		IDAT << "IDAT";
-		std::ostringstream scanlines{ std::ios::binary };
+		std::ostringstream scanlines(std::ios::binary);
 		for(u32 i = 0; i < height; i++) {
 			scanlines.write("\x00", 1); // Null predictor
 			scanlines.write((const char*) data + width * 4 * i, width * 4);

--- a/src/util/png.cpp
+++ b/src/util/png.cpp
@@ -37,11 +37,11 @@ static void writeChunk(std::ostringstream &target, const std::string &chunk_str)
 
 std::string encodePNG(const u8 *data, u32 width, u32 height, s32 compression)
 {
-	auto file = std::ostringstream(std::ios::binary);
+	std::ostringstream file{ std::ios::binary };
 	file << "\x89PNG\r\n\x1a\n";
 
 	{
-		auto IHDR = std::ostringstream(std::ios::binary);
+		std::ostringstream IHDR{ std::ios::binary };
 		IHDR << "IHDR";
 		writeU32(IHDR, width);
 		writeU32(IHDR, height);
@@ -51,9 +51,9 @@ std::string encodePNG(const u8 *data, u32 width, u32 height, s32 compression)
 	}
 
 	{
-		auto IDAT = std::ostringstream(std::ios::binary);
+		std::ostringstream IDAT{ std::ios::binary };
 		IDAT << "IDAT";
-		auto scanlines = std::ostringstream(std::ios::binary);
+		std::ostringstream scanlines{ std::ios::binary };
 		for(u32 i = 0; i < height; i++) {
 			scanlines.write("\x00", 1); // Null predictor
 			scanlines.write((const char*) data + width * 4 * i, width * 4);

